content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md

```diff
@@ -220,6 +220,22 @@ Unable to connect to the server: x509: certificate signed by unknown authority (
   sudo chown $(id -u):$(id -g) $HOME/.kube/config
   ```
 
+## Kubelet client certificate rotation fails {#kubelet-client-cert}
+
+By default, kubeadm configures a kubelet with automatic rotation of the client certificate located
+in `/var/lib/kubelet/pki/` and path specified in `/etc/kubernetes/kubelet.conf`. If this rotation process
+fails you might see errors such as `x509: certificate has expired or is not yet valid` in kube-apserver logs.
+To fix the issue, first check that `/var/lib/kubelet/pki/kubelet-client-current.pem`
+is a symlink. If not, backup and delete it. The kubelet will recreate it.
+Then follow these steps:
+1. Backup and delete `/etc/kubernetes/kubelet.conf` and regenerate it with
+`kubeadm init phase kubeconfig kubelet`. Pass the original `--config` file you used with `kubeadm init`.
+If you don't have it, manually edit the `kubelet.conf`.
+2. Restart the kubelet (`systemctl restart kubelet`) and wait for
+`kubelet-client-current.pem` to be recreated.
+3. Run `kubeadm init phase kubelet-finalize all` using `--config` file if necessary.
+4. Restart the kubelet again.
+
```

Comment:
i realized this would not work for worker nodes.
here is the updated version. i've made some other updates too.

(note: the `1.` list works like so)

```suggestion
By default, kubeadm configures a kubelet with automatic rotation of client certificates by using the `/var/lib/kubelet/pki/kubelet-client-current.pem` symlink specified in `/etc/kubernetes/kubelet.conf`.
If this rotation process fails you might see errors such as `x509: certificate has expired or is not yet valid`
in kube-apserver logs. To fix the issue you must follow these steps:

1. Backup and delete `/etc/kubernetes/kubelet.conf` and `/var/lib/kubelet/pki/kubelet-client*` from the failed node.
1. From a working control plane node in the cluster that has `/etc/kubernetes/pki/ca.key` execute
`kubeadm kubeconfig user --org system:nodes --client-name system:node:$NODE > kubelet.conf`.
`$NODE` must be set to the name of the existing failed node in the cluster.
Modify the resulted `kubelet.conf` manually to adjust the cluster name and server endpoint,
or pass `kubeconfig user --config` (it accepts `InitConfiguration`). If your cluster does not have
the `ca.key` you must sign the embedded certificates in the `kubelet.conf` externally.
1. Copy this resulted `kubelet.conf` to `/etc/kubernetes/kubelet.conf` on the failed node.
1. Restart the kubelet (`systemctl restart kubelet`) on the failed node and wait for
`/var/lib/kubelet/pki/kubelet-client-current.pem` to be recreated.
1. Run `kubeadm init phase kubelet-finalize all`. This will make the new `kubelet.conf` file use `/var/lib/kubelet/pki/kubelet-client-current.pem` and will restart the kubelet.
1. Make sure the node becomes `Ready`.
```